templates/zlogin

```diff
@@ -8,52 +8,38 @@
 #
 
 (
-  # Function to determine the need of a zcompile. If the .zwc file
-  # does not exist, or the base file is newer, we need to compile.
-  # These jobs are asynchronous, and will not impact the interactive shell
-  zcompare() {
-    if [[ -s ${1} && ( ! -s ${1}.zwc || ${1} -nt ${1}.zwc ) ]]; then
-      zcompile ${1}
-    fi
-  }
-
   local zim_mods=${ZIM_HOME}/modules
   setopt EXTENDED_GLOB
+  autoload -U zrecompile
 
   # zcompile the completion cache; siginificant speedup.
   for file in ${ZDOTDIR:-${HOME}}/.zcomp^(*.zwc)(.); do
-    zcompare ${file}
+    zrecompile -pq ${file}
   done
```

Comment:
I see the `.zcompdump` file is already being compiled in `modules/completion/init.zsh`, and it's considering the `zcompdump_file` configuration variable there, which is a more complete solution

Reply:
Can I remove this block? See #217